madara/crates/client/rpc/src/lib.rs

```diff
@@ -108,6 +122,43 @@ impl Starknet {
             .or_internal_server_error("Error getting L1 last confirmed block")?
             .unwrap_or_default())
     }
+
+    pub async fn sync_status(&self) -> StarknetRpcResult<StarknetSyncStatus> {
```

Comment:
idk if it's worth having a struct inside Starknet directly for that, the logic could be directly inside of the endpoint function
(StarknetSyncStatus would go away too in that case)